### ISSUE TITLE
Accessibility feedback on the Spacer Block

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#30157]
**What problem does this address?**

In chatting with Taylor Ardnt a few weeks ago around using FSE with screen reader technology, the Spacer Block came up as a particular problem point in the current FSE and general editing experience. From that conversation, two issues in particular came up that I wanted to flag here:

- Ensure the Spacer Block description is properly read out when using VoiceOver. 
- Scale back the number of Spacer Blocks included in the default template when one enables FSE.

**What other feedback to people have around how the Spacer Block can be improved from an accessibility point of view?**

It would be great to gather feedback here to then figure out specific enhancements can be made.

**Comment by carolinan** [comment]
We can reduce the spacer blocks in TT1 Blocks after
https://github.com/WordPress/gutenberg/pull/29779
And / Or https://github.com/WordPress/gutenberg/issues/28845

**Comment by joedolson** [comment]
For a general reference, the spacer block needs to serve very different purposes depending on whether it's viewed from the perspective of the editing process or in the front-end UI of the site. 

In the front-end of the site, it needs to be totally silent: it should be hidden from screen readers and should never contain content of any kind - including CSS content using `:before` or `:after`. This is basically the equivalent of the invisible spacer.gif used in web design in the late 1990's, and should be invisible to the user.

In the back-end of the site, it needs to be perceivable as to purpose; it needs to declare that it's a spacer and any other relevant characteristics, such as the height of the spacer.

**Comment by joedolson** [comment]
Looks like the current output HTML for the spacer block is variations on

`<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>`

Adding `aria-hidden=true` is unnecessary, and could cause problems in some admittedly rare circumstances. If I hypothesize developers abusing the element to insert content or interactions via JS (for example, using a spacer div as a container to load feedback about the content above or below), then the `aria-hidden=true` attribute will make that content inaccessible. While this is a rare circumstance, I don't see any way that using `aria-hidden` benefits the user. An empty `div` is always going to be a silent presentational element, as it contains no text content, has no role, and no functionality.

**Comment by joedolson** [comment]
Block is announced for me in the admin as 'Block: Spacer'. I think it might be useful to also announce the height of a given spacer. It would be a very small amount of additional text announced, but would help in creating a mental map of the output. E.g., "Block: Spacer - 200px tall".  I'd appreciate a comment from @alexstine  or @MarcoZehe regarding whether they'd consider that useful, however.

**Comment by alexstine** [comment]
I am not sure that would be useful considering I am not sure what 200px really is anymore.

**Comment by Bovelett** [comment]
@annezazu @joedolson I'm softly pulling your sleeves here. The fact that it needs to serve two purposes depending on where the element is perceived (backend or frontend), does that make this impossible to solve? I'm asking because if that is the case, I think maybe we should put some information about the in the documentation? Before I head over to the documentation team, I would be grateful to hear your thoughts.

[https://wordpress.org/documentation/article/spacer-block/ ](url)

**Comment by annezazu** [comment]
I defer to @joedolson and others on the accessibility team as the best way forward. From there, we would dev efforts to implement but keep in mind that we are also very low on collective development time.

**Comment by Infinite-Null** [comment]
I’d love to help out if there’s anything actionable here. Please let me know how I can contribute!

**Comment by joedolson** [comment]
I think the main question here is "what is the outstanding issue?"

Based on the original report, I think that the problem was specific to VoiceOver/Mac not reporting the name of the block correctly. But that doesn't include any details on *how* it wasn't being reported correctly. 

As I look at FSE today, the Spacer Block is read as "Block: spacer, empty document". This is because it has `role="document"`, and no content. There's an open issue discussing the usage of `role="document"`, but that's a much larger, complicated topic. (See https://github.com/WordPress/gutenberg/issues/42158). 

Without more information about the problem in VoiceOver, as far as I can tell this is fine. Adding additional information, such as the height of the spacer, could be considered an enhancement, but is not necessary; that information is discoverable in the block settings, and is not critical to understanding the content.
